pkg/services/alerting/conditions/query_interval_test.go

```diff
@@ -0,0 +1,159 @@
+package conditions
+
+import (
+	"context"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/services/validations"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/services/alerting"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+// the time-range is 5m (300seconds) for every test-case,
+// maxDataPoints is 1500 for every test-case,
+// so the interval for this simple case should be 300s/1500 = 200ms,
+// but in some cases this is overridden by dashboard-panel-min-interval
+// or by datasource-min-interval
+
+func TestQueryInterval(t *testing.T) {
+	Convey("When evaluating query condition, regarding the interval value", t, func() {
+		Convey("Can handle interval-calculation with no panel-min-interval and no datasource-min-interval", func() {
+			// no panel-min-interval in the queryModel
+			queryModel := `{"target": "aliasByNode(statsd.fakesite.counters.session_start.mobile.count, 4)"}`
+
+			// no datasource-min-interval
+			var dataSourceJson *simplejson.Json = nil
+
+			verifier := func(query plugins.DataSubQuery) {
+				So(query.IntervalMS, ShouldEqual, 200)
+				So(query.MaxDataPoints, ShouldEqual, 1500)
```

Comment:
Can you replace 1500 with `interval.DefaultRes`? In the other tests as well?

Reply:
done